sklearn/metrics/classification.py

```diff
@@ -1239,8 +1245,12 @@ def _check_set_wise_labels(y_true, y_pred, average, labels, pos_label):
                                      "%r" % (pos_label, present_labels))
             labels = [pos_label]
         else:
+            applicable_options = list(average_options)
+            if y_type == 'multiclass':
+                applicable_options = applicable_options.remove('samples')
```

Comment:
should be ``applicable_options.remove('samples')`` :)
I think we can use average_options directly because it's not used later.